omc3/optics_measurements/crdt.py

```diff
@@ -0,0 +1,215 @@
+    line[1], line[2] = -crdt['line'][1], -crdt['line'][2]
+    conj_translate_line = translate_line_to_col(line)
+
+    if all(elem in joined_df[0].columns for elem in [translate_line[AMPLITUDE], translate_line[PHASE]]):
+        return translate_line, 1
+    if all(elem in joined_df[0].columns for elem in [conj_translate_line[AMPLITUDE], conj_translate_line[PHASE]]):
+        return conj_translate_line, -1
+    raise ValueError(f"No data for line {crdt['line']} found.")
+
+
+def translate_line_to_col(line):
+    plane = line[0]
+    line = f"{line[1]}{line[2]}".replace('-', '_')
+
+    return dict(zip(CRDT_COLUMNS, [f'{COL_AMP}{line}_{plane}',
+                                   f'{COL_ERR}{COL_AMP}{line}_{plane}',
+                                   f'{COL_PHASE}{line}_{plane}',
+                                   f'{COL_ERR}{COL_PHASE}{line}_{plane}']))
+
+
+def joined_planes(input_files):
+    """
+    Merges DataFrame from the two planes in one df
+    Parameters:
+        input_files
+    Returns:
+        merged DataFrame
+    """
+    joined_dfs = []
+
+    assert len(input_files['X']) == len(input_files['Y'])
+
+    for linx, liny in zip(input_files['X'], input_files['Y']):
+
+        for df, plane in zip((linx, liny), PLANES):
+            rename_cols(df, plane, ['NAME', 'S',
+                                    f'{COL_TUNE}{plane}', f'{COL_AMP}{plane}',
+                                    f'{COL_MU}{plane}', f'{COL_MU}{plane}SYNC'])
+
+        joined_dfs.append(pd.merge(left=linx,
+                                   right=liny,
+                                   on=['NAME', 'S'],
+                                   how='inner',
+                                   sort=False,
+                                   suffixes=(False, False)
+                                   ).set_index('NAME'))
+
+    return joined_dfs
+
+
+def rename_cols(df, suffix, exceptions=['']):
+    df.columns = [f'{col}_{suffix}' if col not in exceptions else col for col in df.columns]
+    return df
+
+
+def get_crdt_phases(joined_dfs, crdt, lines_and_phases, phase_sign):
+    phases, err_phases = np.zeros((len(joined_dfs), len(joined_dfs[0]))), np.zeros((len(joined_dfs), len(joined_dfs[0])))
+
+    for idx, joined_df in enumerate(joined_dfs):
+        phases[idx, :] = (phase_sign*joined_df[lines_and_phases[PHASE]] -
+                          crdt['line'][1]*joined_df['MUX'] -
+                          crdt['line'][2]*joined_df['MUY'] -
+                          (3 - (np.abs(crdt['line'][1])+np.abs(crdt['line'][2])-1)//2)/2*np.pi)
+        try:
+            err_phases[idx, :] = np.sqrt(joined_df[lines_and_phases[f'{ERR}{PHASE}']]**2+
+                                         (crdt['line'][1]*joined_df['ERRMUX'])**2+
+                                         (crdt['line'][2]*joined_df['ERRMUY'])**2)
+        except KeyError:
+            err_phases[idx, :] = np.NaN
+
+    return np.mean(phases, axis=0), np.nanmean(err_phases, axis=0)
```

Comment:
I would say, a circular mean should be used, and error should also take into account spread of measured phase, for instance if the measured phases would be uniformly distributed. One has no clue, irrespective of single measurement errors.

Reply:
phases have been fixed and test included